openedx/core/djangoapps/course_roles/models.py

```diff
@@ -0,0 +1,104 @@
+"""
+Models for course roles schema
+"""
+from django.contrib.auth import get_user_model
+from django.db import models
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from organizations.models import Organization
+
+User = get_user_model()
+
+
+class CourseRolesRole(models.Model):
+    """
+    Model for a course roles role.
+
+    A role is a collection of permissions that can be assigned to a user.
+    The services field defines in which service UI the role is intended to be assigned, such as CMS and/or LMS.
+
+    """
+    name = models.CharField(max_length=255)
+    services = models.ManyToManyField('CourseRolesService', through='CourseRolesRoleService')
+    permissions = models.ManyToManyField('CourseRolesPermission', through='CourseRolesRolePermissions')
+    users = models.ManyToManyField(User, through='CourseRolesUserRole')
+
+    def __str__(self):
+        return self.name
+
+
+class CourseRolesPermission(models.Model):
+    """
+    Model for a course roles permission.
+
+    A permission represents what a user can do.
+    """
+    name = models.CharField(max_length=255, unique=True)
+
+    def __str__(self):
+        return self.name
+
+
+class CourseRolesRolePermissions(models.Model):
```

Comment:
FWIW, I think it's fine to dispense with the `CourseRoles` prefix for all these models. You don't really need the namespacing, and I think it'd make them easier to read.